File: framework/web/src/main/java/top/bettercode/simpleframework/config/CaffeineCacheProperties.java
package top.bettercode.simpleframework.config;

import com.github.benmanes.caffeine.cache.CaffeineSpec;
import java.util.Map;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * @author Peter Wu
 */
@ConfigurationProperties(prefix = "summer.cache.caffeine")
public class CaffeineCacheProperties {

  /**
   * 存储缓存配置,key:name ,value:spec {@link CaffeineSpec}
   */
  private Map<String, String> caches;
  /**
   * 默认缓存配置
   */
  private String defaultSpec = "maximumSize=1000,expireAfterAccess=10s";

  public Map<String, String> getCaches() {
    return caches;
  }

  public CaffeineCacheProperties setCaches(Map<String, String> caches) {
    this.caches = caches;
    return this;
  }

  public String getDefaultSpec() {
    return defaultSpec;
  }

  public CaffeineCacheProperties setDefaultSpec(String defaultSpec) {
    this.defaultSpec = defaultSpec;
    return this;
  }
}

File: web/src/main/kotlin/top/bettercode/summer/web/properties/CaffeineCacheProperties.kt
package top.bettercode.summer.web.properties

import org.springframework.boot.context.properties.ConfigurationProperties

/**
 * @author Peter Wu
 */
@ConfigurationProperties(prefix = "summer.cache.caffeine")
open class CaffeineCacheProperties {
    /**
     * 存储缓存配置,key:name ,value:spec [com.github.benmanes.caffeine.cache.CaffeineSpec]
     */
    var caches: Map<String, String>? = null


    /**
     * 默认缓存配置
     */
    var defaultSpec = "maximumSize=1000,expireAfterAccess=10s"

}
